index.js

```diff
@@ -407,107 +407,137 @@ function buildInnerObject (context, location) {
   return code
 }
 
-function mergeAllOfSchema (context, location, schema, mergedSchema) {
-  const allOfLocation = location.getPropertyLocation('allOf')
+function mergeAllOfSchema (context, location) {
+  const schema = location.schema
+  const schemaWithoutAllOf = clone(schema)
+  const clonedAllOfSchemas = schemaWithoutAllOf.allOf
 
-  for (let i = 0; i < schema.allOf.length; i++) {
-    let allOfSchema = schema.allOf[i]
+  delete schemaWithoutAllOf.allOf
 
-    if (allOfSchema.$ref) {
-      const allOfSchemaLocation = allOfLocation.getPropertyLocation(i)
-      allOfSchema = resolveRef(context, allOfSchemaLocation, allOfSchema.$ref).schema
-    }
+  const allOfSchemasLocation = location.getPropertyLocation('allOf')
+  allOfSchemasLocation.schema = clonedAllOfSchemas
 
-    let allOfSchemaType = allOfSchema.type
-    if (allOfSchemaType === undefined) {
-      allOfSchemaType = inferTypeByKeyword(allOfSchema)
-    }
+  const locations = [
+    new Location(
+      schemaWithoutAllOf,
+      location.schemaId,
+      location.jsonPointer
+    )
+  ]
 
-    if (allOfSchemaType !== undefined) {
-      if (
-        mergedSchema.type !== undefined &&
-        mergedSchema.type !== allOfSchemaType
-      ) {
-        throw new Error('allOf schemas have different type values')
-      }
-      mergedSchema.type = allOfSchemaType
-    }
+  for (let i = 0; i < clonedAllOfSchemas.length; i++) {
+    const allOfSchemaLocation = allOfSchemasLocation.getPropertyLocation(i)
+    locations.push(allOfSchemaLocation)
+  }
 
-    if (allOfSchema.format !== undefined) {
-      if (
-        mergedSchema.format !== undefined &&
-        mergedSchema.format !== allOfSchema.format
-      ) {
-        throw new Error('allOf schemas have different format values')
-      }
-      mergedSchema.format = allOfSchema.format
-    }
+  const mergedLocation = mergeSchemas(context, locations)
+  return mergedLocation
+}
 
-    if (allOfSchema.nullable !== undefined) {
-      if (
-        mergedSchema.nullable !== undefined &&
-        mergedSchema.nullable !== allOfSchema.nullable
-      ) {
-        throw new Error('allOf schemas have different nullable values')
-      }
-      mergedSchema.nullable = allOfSchema.nullable
-    }
+function mergeSchemas (context, locations) {
+  const mergedSchema = {}
 
-    if (allOfSchema.properties !== undefined) {
-      if (mergedSchema.properties === undefined) {
-        mergedSchema.properties = {}
-      }
-      Object.assign(mergedSchema.properties, allOfSchema.properties)
-    }
+  for (let location of locations) {
```

Comment:
Won't a simple for loop be faster than iterators?

It was the case last time I checked

Reply:
This kind of change doesn't create any perf difference unless it places in a _very_ _very_ high-loaded bottleneck (not this library). This library could be optimized by reducing the amount of time it clones the schemas.

Feel free to run benchmarks and prove me wrong.

Reply:
I see, no problem I'll take your word 😄